test/base/builders/SolverOperationBuilder.sol

```diff
@@ -0,0 +1,115 @@
+        solverOperation.to = to;
+        return this;
+    }
+
+    function withValue(uint256 value) public returns (SolverOperationBuilder) {
+        solverOperation.value = value;
+        return this;
+    }
+
+    function withGas(uint256 gas) public returns (SolverOperationBuilder) {
+        solverOperation.gas = gas;
+        return this;
+    }
+
+    function withMaxFeePerGas(uint256 maxFeePerGas) public returns (SolverOperationBuilder) {
+        solverOperation.maxFeePerGas = maxFeePerGas;
+        return this;
+    }
+
+    function withDeadline(uint256 deadline) public returns (SolverOperationBuilder) {
+        solverOperation.deadline = deadline;
+        return this;
+    }
+
+    function withSolver(address solver) public returns (SolverOperationBuilder) {
+        solverOperation.solver = solver;
+        return this;
+    }
+
+    function withControl(address control) public returns (SolverOperationBuilder) {
+        solverOperation.control = control;
+        return this;
+    }
+
+    function withUserOpHash(bytes32 userOpHash) public returns (SolverOperationBuilder) {
+        solverOperation.userOpHash = userOpHash;
+        return this;
+    }
+
+    function withUserOpHash(UserOperation memory userOperation) public returns (SolverOperationBuilder) {
+        solverOperation.userOpHash = userOperation.getUserOperationHash();
+        return this;
+    }
+
+    function withBidToken(address bidToken) public returns (SolverOperationBuilder) {
+        solverOperation.bidToken = bidToken;
+        return this;
+    }
+
+    function withBidToken(
+        address control,
+        UserOperation memory userOperation
+    )
+        public
+        returns (SolverOperationBuilder)
+    {
+        solverOperation.bidToken = IDAppControl(control).getBidFormat(userOperation);
+        return this;
+    }
+
+    function withBidToken(UserOperation memory userOperation) public returns (SolverOperationBuilder) {
+        solverOperation.bidToken = IDAppControl(solverOperation.control).getBidFormat(userOperation);
+        return this;
+    }
+
+    function withBidAmount(uint256 bidAmount) public returns (SolverOperationBuilder) {
+        solverOperation.bidAmount = bidAmount;
+        return this;
+    }
+
+    function withData(bytes memory data) public returns (SolverOperationBuilder) {
+        solverOperation.data = data;
+        return this;
+    }
+
+    function withSignature(bytes memory signature) public returns (SolverOperationBuilder) {
+        solverOperation.signature = signature;
+        return this;
+    }
+
+    function sign(address atlasVerification, uint256 privateKey) public returns (SolverOperationBuilder) {
+        (uint8 v, bytes32 r, bytes32 s) =
+            vm.sign(privateKey, IAtlasVerification(atlasVerification).getSolverPayload(solverOperation));
+        solverOperation.signature = abi.encodePacked(r, s, v);
+        return this;
+    }
```

Comment:
Same again.

IMO this should take AtlasVerification rather than address so we don't have to de-reference the address here.

This seems to lead to weird reverts that are hard to track down, so IMO we should try resolve everything in the test itself then pass instances rather than addresses when we can.